pkg/controller/mysql/tls/tls.go

```diff
@@ -0,0 +1,106 @@
+package tls
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+
+	"github.com/crossplane-contrib/provider-sql/apis/mysql/v1alpha1"
+	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/go-sql-driver/mysql"
+	"github.com/pkg/errors"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LoadConfig loads the TLS configuration when tls mode is set to custom
+func LoadConfig(ctx context.Context, kube client.Client, mode *string, cfg *v1alpha1.TLSConfig) error {
+	if mode == nil || *mode != "custom" {
+		if cfg != nil {
+			return fmt.Errorf("tlsConfig is allowed only when tls=custom")
+		}
+		return nil
+	}
+
+	if err := validateTLSConfig(cfg); err != nil {
+		return err
+	}
+
+	return registerTLS(ctx, kube, cfg)
+}
+
+func validateTLSConfig(cfg *v1alpha1.TLSConfig) error {
+	if cfg == nil ||
+		cfg.CACert.SecretRef.Name == "" ||
+		cfg.CACert.SecretRef.Key == "" ||
+		cfg.ClientCert.SecretRef.Name == "" ||
+		cfg.ClientCert.SecretRef.Key == "" ||
+		cfg.ClientKey.SecretRef.Name == "" ||
+		cfg.ClientKey.SecretRef.Key == "" {
+		return fmt.Errorf("tlsConfig is required when tls=custom")
+	}
+	return nil
+}
+
+func registerTLS(ctx context.Context, kube client.Client, cfg *v1alpha1.TLSConfig) error {
+	if cfg == nil {
+		return nil
+	}
+
+	caCert, err := getSecret(ctx, kube, cfg.CACert.SecretRef)
+	if err != nil {
+		return fmt.Errorf("cannot get CA certificate: %w", err)
+	}
+
+	pool := x509.NewCertPool()
+	if ok := pool.AppendCertsFromPEM(caCert); !ok {
+		return fmt.Errorf("cannot append CA certificate to pool")
+	}
+
+	keyPair, err := getClientKeyPair(ctx, kube, cfg)
+	if err != nil {
+		return err
+	}
+
+	return mysql.RegisterTLSConfig("custom", &tls.Config{
```

Comment:
Looks like "custom" is a global key here. I think this must have a unique name per DB if you connect to two databases?